### PR TITLE
Redo image location calculation

### PR DESCRIPTION
**Pull request by kyryloonufriiev**
I focused more on the screenshoter since this should replace everything else in some point. It's funny how good this logic fits there.

### CODE REVIEW

File: packages/eyes-sdk-core/lib/sdk/EyesBase.js [hunk 1]
```diff
@@ -2187,28 +2178,16 @@ class EyesBase {
     const title = await this.getTitle()
     let domUrl = await this.getDomUrl()
     const imageLocation = await this.getImageLocation()
+    console.log(imageLocation)
```
Comment (amitzur) [inline packages/eyes-sdk-core/lib/sdk/EyesBase.js:2181]: can remove this

File: packages/eyes-sdk-core/lib/sdk/EyesBase.js [hunk 2]
```diff
@@ -2130,25 +2130,16 @@ class EyesBase {
...
-    // END NOTE
-
     // Getting the screenshot (abstract function implemented by each SDK).
     screenshot = await this.getScreenshot()
     this._logger.verbose('Done getting screenshot!')
 
     if (screenshot) {
       // Cropping by region if necessary
       if (!region.isSizeEmpty()) {
+        this._imageLocation = screenshot
```
Comment (amitzur) [inline packages/eyes-sdk-core/lib/sdk/EyesBase.js:2140]: this is a copy of the inside of `getSubScreenshot`, right?

File: packages/screenshoter/src/takeStitchedScreenshot.js [hunk 3]
```diff
@@ -124,7 +124,7 @@ async function takeStitchedImage({
   }
 
   await saveScreenshot(composition, {path: debug.path, name: 'stitched', logger})
-  return composition
+  return {image: composition, region: cropRegion}
```
Comment (amitzur) [inline packages/screenshoter/src/takeStitchedScreenshot.js:127]: shouldn't it be `{image, location}`. What does width/height add here?

File: packages/eyes-sdk-core/lib/sdk/EyesBase.js (hunk 2 above)
Comment (kyryloonufriiev) [reply packages/eyes-sdk-core/lib/sdk/EyesBase.js:2140]: That's right

File: packages/screenshoter/src/takeStitchedScreenshot.js (hunk 3 above)
Comment (kyryloonufriiev) [reply packages/screenshoter/src/takeStitchedScreenshot.js:127]: I want to return the whole effective region, we already have this information, so it doesn't hurt to return it.

**Comment by kyryloonufriiev** [comment]
@amitzur that's correct after taking a screenshot we move the whole page back to its positions before screenshots were taken. And then we taking a dom snapshot.

**Pull request merged**